### ISSUE TITLE
Heap/Memory corruption on Nim 1.4.0 and ARC resolved by accessor change

### ISSUE THREAD

**Issue by elcritch** [status-im/nim-json-rpc#87]
I'm using a fork of your code code, however, tracking down a random memory/heap corruption it seems to track to this line which is the same in both versions:

https://github.com/status-im/nim-json-rpc/blob/99455437ba3d83d5af1c38007fedeeff295e959e/json_rpc/jsonmarshal.nim#L240

I thought I'd share since it appears the same. That line also produces a compiler warning: 
```sh
main/nimble/pkgs/nesper-#devel/nesper/servers/rpc/marshal.nim(218, 20) Warning: cannot prove that field 'params.elems' is accessible [ProveField]
          `jsonIdent`.elems[`pos`]
```

Changing the code from: 
```nim
    let
      paramName = $paramIdent
      jsonElement = quote do:
          `jsonIdent`.elems[`pos`]
```
to:
```nim
    let
      paramName = $paramIdent
      jsonElement = quote do:
        `jsonIdent`[`pos`]
```

Resolves both the compiler warning and stops the semi-random heap corruptions on my fork of the code.

**Comment by jangko** [comment]
Looks like a Nim compiler bug to me. accessing public field directly should not trigger warning like that.

**Comment by elcritch** [comment]
Ok, it wasn't clear to me if it was proper behavior or not. It looks like the code checks it's an JArray first. I'll re-file this as an issue on the compiler. I'll let you all decide to close this issue or not since it does cause RPC failures. It might be helpful for others.

**Comment by jangko** [comment]
fixed by #178, works with nim devel and nim v2
